mod-pubsub-server/src/main/java/org/folio/services/impl/KafkaConsumerServiceImpl.java

```diff
@@ -131,31 +138,50 @@ public KafkaConsumerServiceImpl(@Autowired Vertx vertx,
         } else {
           subscribers
-            .forEach(subscriber -> doRequest(event.getEventPayload(), subscriber.getSubscriberCallback(), HttpMethod.POST, params)
-              .onComplete(getEventDeliveredHandler(event, params.getTenantId(), subscriber)));
+            .forEach(subscriber -> {
+              retry.put(subscriber, new AtomicInteger(0));
+              futureList.add(doRequest(event.getEventPayload(), subscriber.getSubscriberCallback(), HttpMethod.POST, params)
+                .onComplete(getEventDeliveredHandler(event, params.getTenantId(), subscriber, params, retry)));
+            });
         }
-        return Future.succeededFuture();
+        CompositeFuture.all(futureList)
+          .onComplete(ar -> result.complete());
+        return result.future();
       });
   }
 
-  protected Handler<AsyncResult<HttpResponse<Buffer>>> getEventDeliveredHandler(Event event, String tenantId, MessagingModule subscriber) {
+  protected Handler<AsyncResult<HttpResponse<Buffer>>> getEventDeliveredHandler(Event event, String tenantId, MessagingModule subscriber, OkapiConnectionParams params, Map<MessagingModule, AtomicInteger> retry) {
+    retry.get(subscriber).incrementAndGet();
     return ar -> {
       if (ar.failed()) {
         String errorMessage = format("%s event with id '%s' was not delivered to %s", event.getEventType(), event.getId(), subscriber.getSubscriberCallback());
         LOGGER.error(errorMessage, ar.cause());
         auditService.saveAuditMessage(constructJsonAuditMessage(event, tenantId, AuditMessage.State.REJECTED, errorMessage));
+        retryDelivery(event, subscriber, params, retry);
       } else if (ar.result().statusCode() != HttpStatus.HTTP_OK.toInt()
         && ar.result().statusCode() != HttpStatus.HTTP_CREATED.toInt()
         && ar.result().statusCode() != HttpStatus.HTTP_NO_CONTENT.toInt()) {
         String error = format("Error delivering %s event with id '%s' to %s, response status code is %s, %s",
           event.getEventType(), event.getId(), subscriber.getSubscriberCallback(), ar.result().statusCode(), ar.result().statusMessage());
         LOGGER.error(error);
         auditService.saveAuditMessage(constructJsonAuditMessage(event, tenantId, AuditMessage.State.REJECTED, error));
+        retryDelivery(event, subscriber, params, retry);
       } else {
         LOGGER.info("Delivered {} event with id '{}' to {}", event.getEventType(), event.getId(), subscriber.getSubscriberCallback());
         auditService.saveAuditMessage(constructJsonAuditMessage(event, tenantId, AuditMessage.State.DELIVERED));
       }
     };
   }
 
+  private void retryDelivery(Event event, MessagingModule subscriber, OkapiConnectionParams params, Map<MessagingModule, AtomicInteger> retry) {
+    if (retry.get(subscriber).get() <= RETRY_NUMBER) {
+      LOGGER.info("Retry to deliver event {} event with id '{}' to {}", event.getEventType(), event.getId(), subscriber.getSubscriberCallback());
```

Comment:
I think it will be better to change to LOGGER.warn. @OleksiiKuzminov what do you think?